cimr/processor/yamler.py

```diff
@@ -223,8 +223,34 @@ def validate_hash(path, hash):
 def verify_dir(tarred_data):
     """Check directory tree of tarball containing multiple files"""
     for member in tarred_data.getmembers():
-        if not member.name.startswith(DATA_TYPES):
-            logging.error(f' data_type not indicated in dir tree.')
+        # skip sub-directories
+        if member.isdir():
+            continue
+
+        # No any non-file types (links, devices, FIFOs, etc) allowed
+        if not member.isfile():
+            logging.error(f' illegal file type: {member.name}.')
+            sys.exit(1)
+
+        file_path = member.name
+        # Leading '/' in file path NOT allowed
+        if file_path.startswith('/'):
+            logging.error(f' leading / found in archived file.')
```

Comment:
Yes, that is the default behavior of GNU tar. I wrote that in case users may use another tar implementation (such as the one on Solaris or BSD) or non-default option.